Extension/c_cpp_properties.schema.json

```diff
@@ -70,9 +70,22 @@
                         ]
                     },
                     "compileCommands": {
+                        "oneOf": [
+                            {
+                                "type": "string",
+                                "default": ""
```

Comment:
I don't believe that defaults should be defined for these properties since they are optional

Reply:
I added them only because when they exist, the user can clearly see that more than 1 option is available for this type:  
![image](https://github.com/user-attachments/assets/af5ca968-4fed-42d5-a524-d61ebafa576f)  
I can revert this if it breaks a convention you wish to keep...

Reply:
What matters the most is that when we read the value of the property and it is not defined by the user that we read it in as undefined, not some other value.

In package.json when you set a default value, VS Code will return that value to you if the setting is not set by the user. I don't recall off the top of my head the behavior for c_cpp_properties.json. We just need to make sure this works as expected. We will have to trade off having nice IntelliSense for it if it results in undesired behavior when we read the values back.

Reply:
Fair enough, i will double check this both in `cpp_properties.json` and in `settings.json` that we get an `undefined` value if not set.  
Also, previous logic converted empty paths to `undefined`, we can keep this behavior with empty arrays converted to `undefined` as well.

Reply:
@bobbrow I added logic in `CppProperties.parsePropertiesFile()` to:
- handle empty string, empty strings inside the array and empty array to all evaluate to `undefined`.
- a single and non-empty path string will be placed in an array of length 1. 
  
let me know what you think 🙂

Reply:
The approach looks good to me. I believe `parsePropertiesFile` is the right place to force the type to `string[]` so we don't ever have to worry about the ambiguity again.

Reply:
great!  
 just to mention, it's also used in `handleSquiggles()` because we parse `c_cpp_properties.json` there separately.

Reply:
> The approach looks good to me. I believe `parsePropertiesFile` is the right place to force the type to `string[]` so we don't ever have to worry about the ambiguity again.

Quick suggestion, might be an overkill. Instead of forcing the value to be a string array both in `parseCppProperties()` and in `handleSquiggles()`, we create a "version 5" of the properties and add a function "updateToVersion5()" which actually saves a single entry as an array in the json file.   
Again, this might be overkill, just a suggestion.